torchvision/prototype/models/depth/stereo/crestereo.py

```diff
@@ -0,0 +1,1020 @@
+                    # therefore we have to further upscale it by the difference between
+                    # the current level 1 / 16 and the base level 1 / 4.
+                    flow_pred = -upsample_flow(flow_pred_prior, None, factor=scale_to_base)
+                    predictions.append(flow_pred)
+
+                # when constructing the next resolution prior, we resample w.r.t
+                # to the scale of the next level in the pyramid
+                next_resolution = coarse_resolutions[idx + 1]
+                scale_to_next = l_pyramid[next_resolution].shape[2] / flow_pred_prior.shape[2]
+                # we use the flow_up_prior because this is a more accurate estimation of the true flow
+                # due to the convex upsample, which resembles a learned super-resolution module.
+                # this is not necessarily an upsample, it can be a downsample, based on the provided configuration
+                flow_estimates[next_resolution] = -scale_to_next * F.interpolate(
+                    input=flow_pred_prior,
+                    size=l_pyramid[next_resolution].shape[2:],
+                    mode="bilinear",
+                    align_corners=True,
+                )
+
+        # finally we will be doing a full pass through the fine-grained resolution
+        # this coincides with the maximum resolution
+
+        # we keep a separate loop here in order to avoid python control flow
+        # to decide how much iterations should we do based on the current resolution
+        # further more, if provided with an inital flow, there is no need to generate
+        # a prior estimate when moving into the final refinement stage
+
+        for it in range(iterations):
+            use_small_search_patch = (it % 2) == 1
+
+            flow_estimates[max_res] = flow_estimates[max_res].detach()
+            # we run the fine-grained resolution correlations in iterative mode
+            # this means that we are using the fixed window pixel selections
+            # instead of the deformed ones as with the previous steps
+            correlations = self.max_res_correlation_layer(
+                l_pyramid[max_res],
+                r_pyramid[max_res],
+                flow_estimates[max_res],
+                extra_offset=None,
+                use_small_patch=use_small_search_patch,
+                iter_mode=True,
+            )
+
+            net_pyramid[max_res], delta_flow = self.update_block(
+                net_pyramid[max_res], ctx_pyramid[max_res], correlations, flow_estimates[max_res]
+            )
+
+            up_mask = self.mask_predictor(net_pyramid[max_res])
+            flow_estimates[max_res] = flow_estimates[max_res] + delta_flow
+            # at the final resolution we simply do a convex upsample using the base downsample rate
+            flow_pred = -upsample_flow(flow_estimates[max_res], up_mask, factor=self.downsampling_factors[0])
```

Comment:
Just another comment related to `-scale`

Reply:
@YosuaMichael Going through all the `-scale` I agree that it might be the case to try removing them. However looking at the base case which is this final for loop; assuming we receive a valid flow prior for this resolution (cascaded inference) I believe the - might be actually necessary since there is no sign inversion at this stage. Notice that the `flow_estimate[max_res] ` is updated solely with `delta_flow`. My intuition is that it somehow aids with the convex upsampling function.

Reply:
I see, I think we should still have a capability to port the author's weight. In that case, lets keep the `-scale` in my opinion.